### PR TITLE
HIVE-24511: Potential classloader leak in SerDeStorageSchemaReader and add JsonSerde to managed serde

### PR DESCRIPTION
**Pull request by dengzhhu653**

### CODE REVIEW

File: metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java [hunk 1]
```diff
@@ -47,10 +48,10 @@
       Deserializer s = HiveMetaStoreUtils.getDeserializer(conf, tbl, false);
       return HiveMetaStoreUtils.getFieldsFromDeserializer(tbl.getTableName(), s);
     } catch (Exception e) {
-      StringUtils.stringifyException(e);
-      throw new MetaException(e.getMessage());
+      throw new MetaException(StringUtils.stringifyException(e));
```
Comment (belugabehr) [inline metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java:51]: Why does this throw? MetaException?  Try something else?

**Review by belugabehr — CHANGES_REQUESTED** [review]

File: metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java (hunk 1 above)
Comment (dengzhhu653) [reply metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java:51]: If the serde of the table is not a part of `metastore.serdes.using.metastore.for.schema`,  like JsonSerDe(openx).  MetaStore will use the defined serde to get the columns of the table. It's may be better to tell the caller that the exception happens when retrieving the columns from serde other than return null or an empty list, so the GetColumnsOperation can bring the exception to the front.

Comment (dengzhhu653) [reply metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java:51]: Hi, @belugabehr could you please take another look at the changes? thank you

Comment (dengzhhu653) [reply metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java:51]: Hey @belugabehr, could you please take another look? thanks a lot.

**Comment by dengzhhu653** [comment]
close to retrigger the test...

**Pull request closed without merging**